Ash Walker flavor text, balance changes

zxaber opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
- Re-worded Ash Walker flavor text to allow station invasions, but prohibit mass destruction.

- Changed Ash Walker spawns to be limited to one per player.

- Added a 40% chance to gain an extra respawn for every sacrifice made (a player may only have one at a time). The last player to touch the body before it is gibbed is considered the sacrificer.

- Added the ability for the Tendril to revive an Ash Walker, rather than use the body as a sacrifice. This takes five seconds, destroys the old body, and gives them in a new one (keeping their old memories). Should the ghost not be available, the body will simply be sacrificed. Revives do not require, nor do they use, a respawn gained from sacrifices.

- Ash Walkers are no longer able to pilot mechs.

- Fixed borgs being valid for sacrifices. The tendril will still gib them, but not grant any progress to the next egg or roll for a repawn.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
As per the recent vote, updates the flavor text to be less ambiguous. At the same time, since the Antag Freeze is ded, I can also handle some of the concerns that were brought up over the course of me getting feedback.

- Flavor text is currently:

>"Drag the corpses of men and beasts to your nest. It will absorb them to create more of your kind. Invade the strange structure of the outsiders if you must. Do not cause unnecessary destruction, as littering the wastes with ugly wreckage is certain to not gain you favor. Glory to the Necropolis!"

If a better way to word "don't delaminate the engine and plasma flood the station" in an IC way exists, I can change it. As is, I want to allow invasions with a clear restriction on round-ending destruction, and I think this works.

- Blacklisting Ash Walkers from mechs is pretty straightforward. Mechs offer the ability to undo the deliberate weaknesses of Ash Walkers (glass and inability to use guns). It was suggested that I simply blacklist their ability to use mech equipment, but this seems cleaner, and I don't actually see much reason to allow them a battering ram tool.

- Spawn changes were to help limit the ability to just cycle through bodies. This doesn't completely remove it, but instead offers the ability to respawn as a perk for doing sacrificing, which allows for other ghosts to join in. It is set to a random 40% chance because I didn't want to have to keep track of the number of sacrifices made by each player. The tendril respawn system is to help Ash Walkers who are working together to keep eachother alive, which is more important now that respawning is limited.

- Borgs being sacrificial seems like it was an oversight (considering the sacrifice text mentions blood of the corpse covering the eggs). So the tendril can still gib the borg, but it no longer provides any benefit at all (and worse, the borg player might hop back into their brain and yell about how they died).
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
spellcheck: Ash Walker flavor text now allows invasions again, though prohibiting mass destruction.
balance: Ash Walkers can no longer pilot mechs.
balance: Ash Walkers now get only one life per player. However, each sacrifice has a 40% chance to give a bonus spawn (limit of one at a time).
balance: Dead/incapacitated Ash Walkers that are moved to the nest will be given a new body after a few seconds. This does not require an extra spawn, but also does not count as a sacrifice. Ash Walker bodies without a ghost (should the player have moved on to a new spawner) will be sacrificed as normal.
fix: Borgs being gibbed by the Ash Walker tendril will no longer count as a sacrifice, as they do not actually have blood.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

To Do:

- [x] Spell check all the text.
- [x] Fix the merge conflict the PR creation page is already telling me exists.

zxaber committed:
Ashen One

May the flames guide thee

zxaber committed:
Merge branch 'master' into Lethal_Lava_Lizard

XDTM commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -53,16 +53,23 @@
 
+/obj/effect/mob_spawn/human/ash_walker/allow_spawn(mob/user)
+	if(user.key in team.players_spawned)//if player is on exaust list
+		to_chat(user, "<span class='warning'><b>You have exausted your usefulness to the Nechopolis</b>.</span>")
```
```suggestion
		to_chat(user, "<span class='warning'><b>You have exausted your usefulness to the Necropolis</b>.</span>")
```

knacker48 commented:
Good changes overall. I still think limiting of eggs is not needed but hey, you can't please everyone.

Also you say that you have a chance to get a revive and you can only have one, do you give any indicators that you have gained a revive after you sacrifice a body? Like a little message going 'The necropolisis is especially pleased with this sacrifice. You have gained its favor' or something like that?

zxaber committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: XDTM <heliumt@yahoo.it>

zxaber commented:
>Also you say that you have a chance to get a revive and you can only have one, do you give any indicators that you have gained a revive after you sacrifice a body? Like a little message going 'The necropolisis is especially pleased with this sacrifice. You have gained its favor' or something like that?

Yep;
`The Necropolis is pleased with your sacrifice. You feel confident your existence after death is secure.`

HomocidlePancakes commented:
It still spawns eggs for the others, right?

zxaber commented:
Yes, sacrifices still count towards the meat counter and thus extra eggs.

It should be noted that you do actually still require an available egg to use the bonus spawn, too (which is not the case for the revive system; being revived by the tendril has no cost to anyone, and does not use an egg).

Taylorkills2 commented:
Very nice, how does it interact if an ashwalker changes there race or someone who isn't an ashwalker makes sacrifices?
How does it react to people who racechange into an ashwalker?

ATH1909 commented:
Do you keep your memories after using the respawn system, or only after using the revive system?

ExcessiveUseOfCobblestone commented:
Remove policy from flavortext, this should be added to the policy keywords which is on the headmins to reflect.

https://github.com/tgstation/tgstation/pull/44296

ATH1909 commented on code/game/mecha/mecha.dm:
```diff
@@ -878,6 +878,9 @@
 	if(!ishuman(user)) // no silicons or drones in mechas.
 		return
+	if(HAS_TRAIT(user, TRAIT_PRIMITIVE)) //no lavalizards either.
+		to_chat(user, "<span class='warning'>The knowledge to use this device eludes you!</span>")
```
this is a bit awkwardly worded

also, RIP ashwalkers finding and using the abandoned lavaland ripley that sometimes spawns somewhere in lavaland

ATH1909 replied:
Also, I'm not really sure about specifically saying "The knowledge to use this device eludes you!" based on race alone. If someone uses a pride mirror, envy's knife, a mutation toxin, a magic mirror, a changeling's transform ability, etc. to become an ash walker, why would lose their knowledge of how to pilot a mech?

If anything, you should be checking specifically for the ashwalker antag datum, although honestly, if an ash walker manages to find and hijack a mech that hasn't been DNA locked, I say that they've probably earned the right to use it.  Like, if you put a heavy laser cannon in a mech, didn't DNA lock it, and left it in a place where an ash walker could just walk up to it and climb in (or if you left Robotics unsupervised and unattended long enough for an ash walker to create their own mech (and modules for it)), it's entirely YOUR fault that an ash walker got access to a ranged weapon in the form of mech-mounted guns.

Note that wearing a space suit or the like already allows ashwalkers to cover their feetsies, so acquiring an immunity to shardstuns isn't anything particularly special. Note that golems, which also aren't supposed to be able to use most guns, can still pilot mechs and fire mech-mounted guns, and nobody seems to be having a problem with that (even though free golems have the means to create mechs on their own instead of stealing them). Also note that, IIRC, ash walkers can raid the free golem ship to get PKA trigger guard adapters so that they can fire PKAs or use envy's knife to change their race to, say, a skeleton, so even ash walkers getting access to ranged weapons isn't all that special either.

zxaber replied:
Ash Walkers being able to raid the golem ship mean that it's not always the crew's fault that an Ash Walker gets a Durand.

Anyway, the issue of the Pride's mirror has been noted and I'll soon change it to a trait added during the creation of a new Ash Walker via the spawn egg (as a "roundstart" category trait). This means it'll no longer apply to anyone that becomes an Ash Walker by other means and any Ash Walker using the mirror won't suddenly learn mech piloting.

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -53,16 +53,23 @@
 
+/obj/effect/mob_spawn/human/ash_walker/allow_spawn(mob/user)
+	if(user.key in team.players_spawned)//if player is on exaust list
+		to_chat(user, "<span class='warning'><b>You have exausted your usefulness to the Necropolis</b>.</span>")
```
*exhausted

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -43,7 +43,7 @@
 	roundstart = FALSE
 	death = FALSE
-	anchored = FALSE
+	anchored = TRUE
```
👀 

will eggs still fall down chasms created by the destruction of the tendril?

Taylorkills2 replied:
It won't, the ashwalker tendril is special as it only spawns a chasm directly on it's own tile.

zxaber replied:
This was changed because someone expressed concern about Ash Walkers just dragging eggs with them to the station for an easier respawn, which I feel isn't really a healthy mechanic.

Regardless, mechs are anchored and fall into a chasm, so eggs that are above one will fall too.

nemvar replied:
If you do this you should allow people to destroy the eggs somehow. Currently the way to do it is to push them into the chasm.

zxaber replied:
This has been reverted.

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -53,16 +53,23 @@
 /obj/effect/mob_spawn/human/ash_walker/special(mob/living/new_spawn)
 	new_spawn.fully_replace_character_name(null,random_unique_lizard_name(gender))
-	to_chat(new_spawn, "<b>Drag the corpses of men and beasts to your nest. It will absorb them to create more of your kind. Don't leave your nest undefended, protect it with your life. Glory to the Necropolis!</b>")
+	to_chat(new_spawn, "<b>Drag the corpses of men and beasts to your nest. It will absorb them to create more of your kind. Invade the strange structure of the outsiders if you must. Do not cause unnecessary destruction, as littering the wastes with ugly wreckage is certain to not gain you favor. Glory to the Necropolis!</b>")
```
might wanna specify that just kidnapping people from the station and sacrificing them to the nest doesn't count as "unnecessary destruction". sure, it's _implied_, but it's best to be as clear as possible about these sorts of things

zxaber replied:
Seems to me that's handled in the first sentence, "Drag the corpses of men and beast to your nest". I suppose it does say corpses, but every corpse was at one time living.

ExcessiveUseOfCobblestone commented:
Usually when you get a new body you DON'T remember memories. The way it's presented seems like they get made as a new character but it's the same person OOCly, which is surely to cause some confusion.

That said, clarifying that should also be a part of policy :)

ATH1909 commented on code/modules/ruins/objects_and_mobs/ash_walker_den.dm:
```diff
@@ -38,15 +38,33 @@
+			var/mob/living/deliverymob = get_mob_by_key(deliverykey) //mob of said key
+			//there is a 40% chance that the Lava Lizard unlocks their respawn with each sacrifice
+			if(deliverymob && (deliverymob.mind?.has_antag_datum(/datum/antagonist/ashwalker)) && (deliverykey in ashies.players_spawned) && (prob(40)))
+				to_chat(deliverymob, "<span class='warning'><b>The Necopolis is pleased with your sacrifice. You feel confident your existance after death is secure.</b></span>")
```
*existence

ATH1909 commented on code/modules/ruins/objects_and_mobs/ash_walker_den.dm:
```diff
@@ -55,6 +73,17 @@
+	M.update_body()
+	oldmind.transfer_to(M)
+	M.mind.grab_ghost()
+	to_chat(M, "<b>You have been pulled back from beyond the grave, with a new body and renewed purpose. Glory to the Necropolis!</b>")
```
might wanna specifically say that the respawned ash walker keeps their memories

zxaber replied:
I was hoping it would be implied well enough, since the revived Walker gets the same name as their old body as well as the same IC notes, but I could add to the text here a bit.

ATH1909 commented:
Anyway, overall, I really like what you've done with ash walkers in this PR (needing to sacrifice stuff to get your respawn is a great mechanical incentive to follow your role's flavor text/"mission"), I just have a bit of an issue with the mech stuff, as mentioned about.

Also, are you okay with ash walkers killing themselves by the tendril for a full heal? I feel like killing yourself by the tendril as an ash walker could become the new "kill and clone" for them, especially since it doesn't take very long for the tendril to rez you.

ATH1909 commented on code/modules/ruins/objects_and_mobs/ash_walker_den.dm:
```diff
@@ -38,15 +38,33 @@
+				H.notify_ghost_cloning(message = "Your body has been returned to the nest. You are being remade anew!", sound = 'sound/magic/enter_blood.ogg')
+				addtimer(CALLBACK(src, .proc/remake_walker, H.mind, H.real_name), 5 SECONDS)
+				new /obj/effect/gibspawner/generic(get_turf(H))
+				qdel(H)
```
this won't delete any gear that was on the ashwalker's old body, right?

zxaber replied:
No, one of the first things the tendril does is strip bodies of all items.

RaveRadbury commented:
I've said on the forums and I'll reiterate here that I disagree with limiting ashwalker spawns this significantly, however the overall PR is good and I'd prefer trying this whole thing out and adjusting later on rather than see this not go through.

zxaber commented:
@ATH1909 
>Also, are you okay with ash walkers killing themselves by the tendril for a full heal? I feel like killing yourself by the tendril as an ash walker could become the new "kill and clone" for them, especially since it doesn't take very long for the tendril to rez you.

I would love to avoid that if I could, but there's no easy way to stop it. I could restrict the suicide verb, but attacking yourself with a spear works just as well and wouldn't be caught. And even if we watched for self-attacking, you could just have another Walker do it, or even catch an ash storm and wait until soft crit before dragging yourself to the tendril.

Ultimately, I think it's not a big issue, as Lavaland already has other ways to heal. If people want to sit through the suicide, respawn, and re-equip for every single scratch, then it's whatever. The Tendril can't be moved, so they'll have to trek all the way back every time.

Shadowflame909 commented:
It's just balancing the odds between survival pen armor miners use.

Also you only being able to play ash walker once is super lame... A cooldown would suffice as these rounds dont last long.

zxaber commented:
Just sacrifice some fauna. You ought to be doing it anyway to let other ghosts join in, and each sacrifice gives you a 40% chance of getting to take another egg if you die.

bluetshirtguy commented:
I despise ashwalkers, but these changes overall seem like a good compromise.

nemvar commented on code/modules/ruins/objects_and_mobs/ash_walker_den.dm:
```diff
@@ -38,15 +38,33 @@
+				H.gib()
+				visible_message("<span class='notice'>Serrated tendrils eagerly pull [H] apart, but find nothing of interest.</span>")
+				return
+			if(H.mind?.has_antag_datum(/datum/antagonist/ashwalker) && H.get_ghost(FALSE, TRUE)) //special interactions for dead lava lizards with ghosts attached
```
Does this work? Wouldn't this need to be lastmind?

CRITAWAKETS commented:
Honestly, being able to use the tendril for healing is something that ashwalkers need. Currently ashwalkers have no renewable healing sources.

zxaber committed:
update

zxaber committed:
caught it before I got yelled at

zxaber commented:
Changes:
- Ash Walker eggs now exist inside an "eggshell" object (of the same name and sprite). Nothing fundamentally has changed, except the egg can be attacked. It has 80 integrity and breaking it will spawn and promptly gib an ash lizard. Take respawns alongside you during invasions at your own risk.
- As per the above, I removed anchoring eggs that this PR originally had. Abducting Ash Walker eggs sounds funny.
- Spell checked things.

Dennok commented on code/datums/ruins/lavaland.dm:
```diff
@@ -50,7 +50,8 @@
 	suffix = "lavaland_surface_ash_walker1.dmm"
-	cost = 20
+	//cost = 20
+	always_place = TRUE //DEBUG reasons, yell at me if I commit with this still here
```
I yell at you.

ATH1909 commented:
> 80 integrity 

You might want to reduce that to 40 or so, just so that ash walker eggs are easy to break for anyone with, say, a spear.

Still, thanks for making it possible to abduct ash walker eggs!

zxaber committed:
updates

zxaber commented:
New changes:

- Added a new line to the alert when a dead ash lizard has been brought to the nest, that mentions specifically that prior memories are kept. (This is specifically about being revived, not taking a new egg).
- Time to revive is now 20 seconds, to put at least a small delay into getting fellow Walkers back on their feet.
- Revive alert no longer uses notify_ghost_cloning, as that had a "click to re-enter body" message that wouldn't do anything in this case.
- Fixed the revive section not being activated if the player is inside their Ash Walker body (why does get_ghost fail here?).
- Re-added some code I somehow reverted earlier in relation to the egg-per-client limit.

I think with that, this PR is pretty much ready for review.

knacker48 commented:
Wait are you doing the delay on reviving  as an addition to limiting on respawns? Cause you seem to be layering on more and more nerfs as you go along I really don't see why you find it necessary

zxaber commented:
The delay is more to dissuade Walkers from just critting themselves over any minor scratch to get back to full HP. If you've been dead and are being dragged back to the nest, an extra 20 seconds for what amounts to an admin heal (with a skin color change) isn't going to hurt that bad.

Learner10 commented:
If the nest is being invaded, having a 5 second long full-heal would add in road blocks to any invasion.

Anyhow, time to abduct an egg and bring it to Xenobiology.

ATH1909 commented:
You might want to add a message for respawning from an egg (instead of having your old corpse rezzed) that reminds ash lizards that they're only supposed to retain their memories if their old corpse is rezzed, not if they respawn from a new egg.

Alternatively, you could make ash walkers keep the memories of their previous lives (and maybe even their previous ash walker name?) when respawning from an egg, just to make things less confusing/of an administrative headache for admins ("I, uh, randomly wandered in this direction! I totally didn't know beforehand that my previous body's corpse was here!").

kriskog commented:
Merge conflict

zxaber committed:
Merge branch 'master' into Lethal_Lava_Lizard

zxaber commented:
Conflict fixed.

ExcessiveUseOfCobblestone commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -33,6 +33,44 @@
+		yolk.update_body()
+		yolk.gib()
+		qdel(egg)
+	. = ..()
```
```suggestion
	if(!egg)
	    return ..()
	var/mob/living/carbon/human/yolk = new /mob/living/carbon/human/(get_turf(src))
	yolk.fully_replace_character_name(null,random_unique_lizard_name(gender))
	yolk.set_species(/datum/species/lizard/ashwalker)
	yolk.underwear = "Nude"
	yolk.equipOutfit(/datum/outfit/ashwalker)//this is an authentic mess we're making
	yolk.update_body()
	yolk.gib()
	qdel(egg)
	return ..()
```

ExcessiveUseOfCobblestone commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -52,22 +89,36 @@
+	if(user.key in team.players_spawned)//one per person unless you get a bonus spawn
+		to_chat(user, "<span class='warning'><b>You have exhausted your usefulness to the Necropolis</b>.</span>")
+		return FALSE
+	return TRUE
```
```suggestion
	if(!(user.key in team.players_spawned))//one per person unless you get a bonus spawn
	    return TRUE
	to_chat(user, "<span class='warning'><b>You have exhausted your usefulness to the Necropolis</b>.</span>")
	return FALSE
```

ExcessiveUseOfCobblestone commented:
ping when completed

zxaber committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

zxaber committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

zxaber committed:
spaced the spaces

tab good space bad
